sqlmesh/core/context.py

```diff
@@ -1331,51 +1377,60 @@ def _model_tables(self) -> t.Dict[str, str]:
         If a snapshot has not been versioned yet, its view name will be returned.
         """
         return {
-            name: snapshot.table_name()
+            snapshot.fqn: snapshot.table_name()
             if snapshot.version
             else snapshot.qualified_view_name.for_environment(
                 EnvironmentNamingInfo(
                     name=c.PROD, suffix_target=self.config.environment_suffix_target
                 )
             )
-            for name, snapshot in self.snapshots.items()
+            for snapshot in self.snapshots
         }
 
     def _snapshots(
         self, models_override: t.Optional[UniqueKeyDict[str, Model]] = None
-    ) -> t.Dict[str, Snapshot]:
+    ) -> t.Dict[SnapshotId, Snapshot]:
         prod = self.state_reader.get_environment(c.PROD)
         remote_snapshots = (
             {
-                snapshot.name: snapshot
+                snapshot.snapshot_id: snapshot
                 for snapshot in self.state_reader.get_snapshots(prod.snapshots).values()
             }
             if prod
             else {}
         )
+        remote_snapshot_model_fqn_to_snapshot = {
+            snapshot.fqn: snapshot for s_id, snapshot in remote_snapshots.items()
+        }
+        models = models_override or self._models
```

Comment:
Ditto, don't like the idea of using an empty dict to indicate the lack of overrides.